curvefs/src/mds/topology/topology.cpp

```diff
@@ -1257,6 +1235,88 @@ TopoStatusCode TopologyImpl::ChooseSingleMetaServerInZone(
     return TopoStatusCode::TOPO_METASERVER_NOT_FOUND;
 }
 
+TopoStatusCode TopologyImpl::ChooseAvailableMetaServers(
+    std::set<MetaServerIdType> *metaServers,
+    PoolIdType *poolId) {
+    ReadLockGuard rlockMetaserver(metaServerMutex_);
+    std::vector<MetaServer> vec;
+    for (const auto &it : metaServerMap_) {
+        vec.emplace_back(it.second);
+    }
```

Comment:
it seems no need to copy all metaserver from map to vector, directly store metaserver's pointer is meeting current purpose.

Reply:
> it seems no need to copy all metaserver from map to vector, directly store metaserver's pointer is meeting current purpose.

fixed